File: Java(eclipse)/workspace/書籍検索アプリ/src/main/java/todo/model/detail/CsvReader.java
package todo.model.detail;

import todo.model.beans.BookDataBean;
import 自作ライブラリ.TextFileLib;

public class CsvReader {
	
	private BookDataBean[] bookDataBeans;
	
	//----------
	//コンストラクタ
	public CsvReader() {
		
		bookDataBeans = new BookDataBean[0];
	}
	public CsvReader(BookDataBean[] bookDataBeans) {
		
		this.bookDataBeans = bookDataBeans;
	}
	
	//------------
	//CSVを読み込み
	public BookDataBean[] readCsv(String filePath) {
		
		String[][] resultArray  = TextFileLib.readerCsvData(filePath,"Shift-JIS",true);
		bookDataBeans = new BookDataBean[resultArray.length];
		int i = 0;
		for(String[] results : resultArray) {
			bookDataBeans[i] = new BookDataBean(results);
			i++;
		}
		return bookDataBeans;
	}
}

File: C#(VisualStudio)/WorkSpace_ASP.NETCoreMVC/書籍検索アプリ_移植元Java/Models/CSVReader.cs
using 書籍検索アプリ_移植元Java.Models.beans;
using 書籍検索アプリ_移植元Java.自作ライブラリ;

namespace 書籍検索アプリ_移植元Java.Models
{
    public class CSVReader {
	
		private BookDataBean[] bookDataBeans;
	
		//----------
		//コンストラクタ
		public CSVReader() {
		
			bookDataBeans = Array.Empty<BookDataBean>();
		}
		public CSVReader(BookDataBean[] bookDataBeans) {
		
			this.bookDataBeans = bookDataBeans;
		}
	
		//------------
		//CSVを読み込み
		public BookDataBean[] readCsv(string filePath) {
		
			string[][] resultArray  = TextFileLib.ReaderCsvData(filePath,"Shift-JIS",true);
			bookDataBeans = new BookDataBean[resultArray.Length];
			int i = 0;
			foreach(string[] results in resultArray) {
				bookDataBeans[i] = new BookDataBean(results);
				i++;
			}
			return bookDataBeans;
		}
	}
}